server/src/test/java/org/spine3/server/procman/IteratingCommandRouterShould.java

```diff
@@ -0,0 +1,148 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.procman;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import com.google.protobuf.StringValue;
+import org.junit.Before;
+import org.junit.Test;
+import org.spine3.base.Command;
+import org.spine3.base.CommandContext;
+import org.spine3.base.Commands;
+import org.spine3.protobuf.AnyPacker;
+import org.spine3.server.BoundedContext;
+import org.spine3.server.command.CommandBus;
+import org.spine3.server.command.CommandDispatcher;
+import org.spine3.server.type.CommandClass;
+import org.spine3.test.TestCommandFactory;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.spine3.protobuf.Values.newStringValue;
+
+/**
+ * @author Alexander Yevsyukov
+ */
+public class IteratingCommandRouterShould {
+
+    private final TestCommandFactory commandFactory = TestCommandFactory.newInstance(getClass());
+
+    /** The object we test. */
+    private IteratingCommandRouter router;
+
+    /** The command message we route. */
+    private Message sourceMessage;
+
+    /** The context of the command that we route. */
+    private CommandContext sourceContext;
+
+    /** Command messages to be sent. */
+    private final List<Message> messages = ImmutableList.<Message>of(
+            newStringValue("uno"),
+            newStringValue("dos"),
+            newStringValue("tres"),
+            newStringValue("cuatro")
+    );
+
+    @Before
+    public void setUp() {
+        final BoundedContext boundedContext = BoundedContext.newBuilder()
+                                                            .build();
+        final CommandBus commandBus = boundedContext.getCommandBus();
+        commandBus.register(new CommandDispatcher() {
+            @Override
+            public Set<CommandClass> getCommandClasses() {
+                return CommandClass.setOf(StringValue.class);
+            }
+
+            @Override
+            public void dispatch(Command request) throws Exception {
+                // Do nothing.
+            }
+        });
+
+        sourceMessage = newStringValue(getClass().getSimpleName());
+        sourceContext = commandFactory.createCommandContext();
+
+        router = new IteratingCommandRouter(commandBus, sourceMessage, sourceContext);
+        router.addAll(messages);
+    }
+
+    @Test
+    public void routeFirst() throws Exception {
```

Comment:
The test method names are not usual for Spine in this test class. Is it on purpose? If so, what about all the other test methods in other classes?